File: API/Java/ElapsedTime.java
/*##############################################################################################
#                                                                                              #
#     #############       #############       #############       ####                ####     #
#    #             #     #             #     #             #     #    #              #    #    #
#    #    #####    #     #    #########      #    #####    #     #    #              #    #    #
#    #    #   #    #     #    #              #    #   #    #     #    #              #    #    #
#    #    #####    #     #    #              #    #####    #     #    #              #    #    #
#    #             #     #    #########      #             #     #    #              #    #    #
#    #             #     #             #     #             #     #    #              #    #    #
#    #    #####    #      #########    #     #    #####    #     #    #              #    #    #
#    #    #   #    #              #    #     #    #   #    #     #    #              #    #    #
#    #    #   #    #      #########    #     #    #   #    #     #    #########      #    #    #
#    #    #   #    #     #             #     #    #   #    #     #             #     #    #    #
#     ####     ####       #############       ####     ####       #############       ####     #
#                                                                                              #
#   Author: Stefano Rebughini <ste.rebu@outlook.it>                                            #
#                                                                                              #
################################################################################################
#                                                                                              #
#   License                                                                                    #
#                                                                                              #
#   This file is part of ASALI.                                                                #
#                                                                                              #
#   ASALI is free software: you can redistribute it and/or modify                              #
#   it under the terms of the GNU General Public License as published by                       #
#   the Free Software Foundation, either version 3 of the License, or                          #
#   (at your option) any later version.                                                        #
#                                                                                              #
#   ASALI is distributed in the hope that it will be useful,                                   #
#   but WITHOUT ANY WARRANTY; without even the implied warranty of                             #
#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the                              #
#   GNU General Public License for more details.                                               #
#                                                                                              #
#   You should have received a copy of the GNU General Public License                          #
#   along with ASALI. If not, see <http://www.gnu.org/licenses/>.                              #
#                                                                                              #
##############################################################################################*/

import java.io.*; 
import java.util.*;
import java.lang.*;
import java.text.*;

class ElapsedTime 
{
    public static void main(String args[])  throws Exception 
    { 
		int N = Integer.valueOf(args[0]);

        //Set up mixture composition
        ArrayList<String> names = new ArrayList<String>();
		names.add("H2");
		names.add("O2");
		names.add("C3H8");
		names.add("C2H6");
		names.add("CH4");
		names.add("CO2");
		names.add("HE");
		names.add("N2");
		names.add("NH3");
        
        Vector<Double> x = new Vector<Double>();
        x.add(0.1);
        x.add(0.1);
        x.add(0.1);
        x.add(0.1);
        x.add(0.1);
        x.add(0.1);
        x.add(0.1);
        x.add(0.1);
        x.add(0.2);

		//Initialize variables
		double initializationStart = System.nanoTime();
		for (int i=0;i<N-1;i++)
		{
			Asali asali_ = new Asali();
		}
		Asali asali_ = new Asali();
		double initializationEnd = System.nanoTime();
		
		double []  mu, cp, h, s, cond, diff_mix, v, l;
		double [][] diff;
		double MWmix, rho, mumix, cpmassmix, cpmolemix, condmix, hmassmix, hmolemix, smolemix, smassmix;


		double estimationStart = System.nanoTime();
		for (int i=0;i<N;i++)
		{
			asali_.set_pressure(4e05);
			asali_.set_temperature(393.15);
			asali_.set_number_of_species(9);
			asali_.set_species_names(names);
			asali_.set_mole_fraction(x);

			mu        = asali_.get_species_viscosity();
			cp        = asali_.get_species_mass_specific_heat();
			h         = asali_.get_species_mass_enthalpy();
			s         = asali_.get_species_mass_entropy();
			cond      = asali_.get_species_thermal_conductivity();
			diff_mix  = asali_.get_mixture_diffusion();
			v         = asali_.get_aritmetic_mean_gas_velocity();
			l         = asali_.get_mean_free_path();
			diff      = asali_.get_binary_diffusion();
			MWmix     = asali_.get_mixture_molecular_weight();
			rho       = asali_.get_density();
			mumix     = asali_.get_mixture_viscosity();
			cpmassmix = asali_.get_mixture_mass_specific_heat();
			cpmolemix = asali_.get_mixture_molar_specific_heat();
			condmix   = asali_.get_mixture_thermal_conductivity();
			hmassmix  = asali_.get_mixture_mass_enthalpy();
			hmolemix  = asali_.get_mixture_molar_enthalpy();
			smolemix  = asali_.get_mixture_molar_entropy();
			smassmix  = asali_.get_mixture_molar_entropy();
		}
		double estimationEnd = System.nanoTime();
		double estimationTime = (estimationEnd - estimationStart)/(1.0e09*(double)N);
		double initializationTime = (initializationEnd - initializationStart)/(1.0e09*(double)N);
		
		System.out.println("Java version");
		System.out.printf("Initialization (s):  %.3e\n", initializationTime);
		System.out.printf("Calculation (s):     %.3e\n", estimationTime);
    }
}

File: API/Julia/elapsed-time.jl
################################################################################################
#                                                                                              #
#     #############       #############       #############       ####                ####     #
#    #             #     #             #     #             #     #    #              #    #    #
#    #    #####    #     #    #########      #    #####    #     #    #              #    #    #
#    #    #   #    #     #    #              #    #   #    #     #    #              #    #    #
#    #    #####    #     #    #              #    #####    #     #    #              #    #    #
#    #             #     #    #########      #             #     #    #              #    #    #
#    #             #     #             #     #             #     #    #              #    #    #
#    #    #####    #      #########    #     #    #####    #     #    #              #    #    #
#    #    #   #    #              #    #     #    #   #    #     #    #              #    #    #
#    #    #   #    #      #########    #     #    #   #    #     #    #########      #    #    #
#    #    #   #    #     #             #     #    #   #    #     #             #     #    #    #
#     ####     ####       #############       ####     ####       #############       ####     #
#                                                                                              #
#   Author: Stefano Rebughini <ste.rebu@outlook.it>                                            #
#                                                                                              #
################################################################################################
#                                                                                              #
#   License                                                                                    #
#                                                                                              #
#   This file is part of ASALI.                                                                #
#                                                                                              #
#   ASALI is free software: you can redistribute it and/or modify                              #
#   it under the terms of the GNU General Public License as published by                       #
#   the Free Software Foundation, either version 3 of the License, or                          #
#   (at your option) any later version.                                                        #
#                                                                                              #
#   ASALI is distributed in the hope that it will be useful,                                   #
#   but WITHOUT ANY WARRANTY; without even the implied warranty of                             #
#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the                              #
#   GNU General Public License for more details.                                               #
#                                                                                              #
#   You should have received a copy of the GNU General Public License                          #
#   along with ASALI. If not, see <http://www.gnu.org/licenses/>.                              #
#                                                                                              #
################################################################################################

using Printf
include("asali.jl")


species = String["H2","O2","C3H8","C2H6","CH4","CO2","HE","N2","NH3"]
x = Float64[0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.2]

n = parse(Int, ARGS[1])

estimation_start = time_ns()
for i in 1:n
    asali.set_temperature(393.15)
    asali.set_pressure(4e05)
    asali.set_number_of_species(9)
    asali.set_species_names(species)
    asali.set_mole_fraction(x)

    mu = asali.get_species_viscosity()
    MWmix = asali.get_mixture_molecular_weight()
    rho = asali.get_density()
    bdiff = asali.get_binary_diffusion()
    cp = asali.get_species_mass_specific_heat()
    h = asali.get_species_mass_enthalpy()
    s = asali.get_species_mass_entropy()
    cond = asali.get_species_thermal_conductivity()
    condmix = asali.get_mixture_thermal_conductivity()
    mumix = asali.get_mixture_viscosity()
    diffmix = asali.get_mixture_diffusion()
    v = asali.get_aritmetic_mean_gas_velocity()
    l = asali.get_mean_free_path()
    cpmolemix = asali.get_mixture_molar_specific_heat()
    cpmassmix = asali.get_mixture_mass_specific_heat()
    hmolemix = asali.get_mixture_molar_enthalpy()
    hmassmix = asali.get_mixture_mass_enthalpy()
    smolemix = asali.get_mixture_molar_entropy()
    smassmix = asali.get_mixture_mass_entropy()
end

estimation_end = time_ns()

calc_time = (estimation_end - estimation_start) * 1e-9 / n

println("Julia version")
println("Initialization (s):  n.a.")
println("Calculation (s):     ", @sprintf("%.3e", calc_time))